distributed/utils.py

```diff
@@ -371,29 +371,39 @@ def in_async_call(loop, default=False):
         return False
 
 
-def sync(loop, func, *args, callback_timeout=None, **kwargs):
+def sync(
+    loop: IOLoop,
+    func: Callable[..., Awaitable[T] | asyncio.Future[T]],
+    *args: AnyType,
+    callback_timeout: str | float | timedelta | None = None,
+    **kwargs: AnyType,
+) -> T:
     """
     Run coroutine in loop running in separate thread.
     """
-    callback_timeout = _parse_timedelta(callback_timeout, "s")
-    if loop.asyncio_loop.is_closed():
+    timeout = _parse_timedelta(callback_timeout, "s")
+    if loop.asyncio_loop.is_closed():  # type: ignore[attr-defined]
         raise RuntimeError("IOLoop is closed")
 
     e = threading.Event()
     main_tid = threading.get_ident()
-    result = error = future = None  # set up non-locals
+
+    # set up non-locals
+    result: T
+    error = None
+    future: asyncio.Future[T] | asyncio.Task[T]
 
     @gen.coroutine
-    def f():
+    def f() -> Generator[Any, Any, None]:
```

Comment:
Argh I fell into the trap in this file
```suggestion
    def f() -> Generator[AnyType, AnyType, None]:
```

Reply:
Fixed already :)